### PR TITLE
Add the into_stream() combinator

### LINKED ISSUE #52: Method for creating a stream from a future
In futures there is [`stream::once`](https://docs.rs/futures/latest/futures/prelude/stream/fn.once.html) which allows creating a stream from a future. I would argue it generic enough and simple enough to warrant it being functionality provided in this library.

Would a `stream::future` or similar named function be accepted with a PR?

### PR DESCRIPTION
**Pull request by notgull**
Resolves #52

### CODE REVIEW

**Comment by taiki-e** [comment]
Why does this add both freestanding function and method?

**Comment by notgull** [comment]
I assumed that was the status quo for `futures_lite`, like how `zip` and `or` are methods.

**Comment by notgull** [comment]
I assumed that was the status quo for `futures_lite`, like how `zip` and `or` are methods.

**Comment by taiki-e** [comment]
I guess it is only happening with methods that take two or more futures/streams. (e.g., [catch_unwind](https://docs.rs/futures-lite/latest/futures_lite/future/trait.FutureExt.html#method.catch_unwind) is trait method only)
There has been discussion in the past in `futures` that freestanding functions are preferred for such methods: https://github.com/rust-lang/futures-rs/issues/1215

**Comment by taiki-e** [comment]
It seems you have removed the trait method, but I have the impression that `into_` is a prefix that is often used in the method (e.g., `String::into_bytes`, `{BufReader,BufWriter}::into_inner)`.

**Comment by notgull** [comment]
You're right. How would you feel about just the name `stream()`?

**Comment by taiki-e** [comment]
Maybe `stream::from_future`?

That said, I have no strong opinion on this as I tend to prefer the `futures`'s APIs that take the future as an argument (i.e., I think `stream::once` is the ideal name here).

**Comment by taiki-e** [comment]
@smol-rs/admins: Any thoughts on the API here?

**Comment by zeenix** [comment]
> @smol-rs/admins: Any thoughts on the API here?

I agree with you about the naming. `stream::once` seems better to me as well.

File: src/future.rs [hunk 1]
```diff
@@ -25,6 +25,8 @@ use core::fmt;
 use core::marker::PhantomData;
 use core::pin::Pin;
 
+use futures_core::Stream;
+
```
Comment (zseri) [inline src/future.rs:29]: this appears to be unnecessary, right?

**Comment by zseri** [comment]
So this does just the same as https://docs.rs/futures-util/0.3.24/futures_util/future/struct.IntoStream.html ?

**Review by zseri — APPROVED** [review]

**Comment by zseri** [comment]
this should be squash-merged.

**Pull request merged**